src/unix/linux/x11/audio.c

```diff
@@ -9,36 +9,38 @@
 
 #include "dl/libasound.h"
 
-#define AUDIO_SAMPLE_SIZE sizeof(int16_t)
-
-#define AUDIO_BUF_SIZE(ctx) \
-	((ctx)->sample_rate * (ctx)->channels * AUDIO_SAMPLE_SIZE)
-
 struct MTY_Audio {
 	snd_pcm_t *pcm;
 
 	bool playing;
+	MTY_AudioSampleFormat sample_format;
 	uint32_t sample_rate;
 	uint32_t min_buffer;
 	uint32_t max_buffer;
 	uint8_t channels;
+	uint32_t frame_size;
+	uint32_t buffer_size;
 	uint8_t *buf;
 	size_t pos;
 };
 
-MTY_Audio *MTY_AudioCreate(uint32_t sampleRate, uint32_t minBuffer, uint32_t maxBuffer, uint8_t channels,
-	const char *deviceID, bool fallback)
+MTY_Audio *MTY_AudioCreate(const MTY_AudioFormat *format, uint32_t minBuffer,
+	uint32_t maxBuffer, const char *deviceID, bool fallback)
 {
 	if (!libasound_global_init())
 		return NULL;
 
 	MTY_Audio *ctx = MTY_Alloc(1, sizeof(MTY_Audio));
-	ctx->sample_rate = sampleRate;
-	ctx->channels = channels;
+	ctx->sample_format = format->sampleFormat;
+	ctx->sample_rate = format->sampleRate;
+	ctx->channels = format->channels;
+	ctx->frame_size = format->channels *
+		(format->sampleFormat == MTY_AUDIO_SAMPLE_FORMAT_FLOAT ? sizeof(float) : sizeof(int16_t));
+	ctx->buffer_size = format->sampleRate * ctx->frame_size;
 
-	uint32_t frames_per_ms = lrint((float) sampleRate / 1000.0f);
-	ctx->min_buffer = minBuffer * frames_per_ms;
-	ctx->max_buffer = maxBuffer * frames_per_ms;
+	uint32_t samples_per_ms = lrint((float) format->sampleRate / 1000.0f);
+	ctx->min_buffer = minBuffer * samples_per_ms;
+	ctx->max_buffer = maxBuffer * samples_per_ms;
 
 	bool r = true;
```

Comment:
A better point, looking at these; It's basically the same thing every time, so why not a common function?

Reply:
Will consider this as well.

Reply:
This turned out to be a great idea. Refactored into a common function across platforms.